src/Appwrite/Platform/Modules/Proxy/Http/Rules/Create.php

```diff
@@ -0,0 +1,198 @@
+            ->inject('dbForProject')
+            ->callback([$this, 'action']);
+    }
+
+    public function action(string $domain, string $resourceType, string $resourceId, Response $response, Document $project, Certificate $queueForCertificates, Event $queueForEvents, Database $dbForPlatform, Database $dbForProject)
+    {
+        $mainDomain = System::getEnv('_APP_DOMAIN', '');
+        if ($domain === $mainDomain) {
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'You cannot assign your main domain to specific resource. Please use subdomain or a different domain.');
+        }
+
+        $sitesDomain = System::getEnv('_APP_DOMAIN_SITES', '');
+        $functionsDomain = System::getEnv('_APP_DOMAIN_FUNCTIONS', '');
+
+        if ($domain === $functionsDomain || $domain === $sitesDomain) {
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'You cannot assign your functions or sites domain to a specific resource. Please use a different domain.');
+        }
+
+        if ($domain === 'localhost' || $domain === APP_HOSTNAME_INTERNAL) {
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'This domain name is not allowed. Please pick another one.');
+        }
+
+        // TODO: @christyjacob remove once we migrate the rules in 1.7.x
+        if (System::getEnv('_APP_RULES_FORMAT') === 'md5') {
+            $document = $dbForPlatform->getDocument('rules', md5($domain));
+        } else {
+            $document = $dbForPlatform->findOne('rules', [
+                Query::equal('domain', [$domain]),
+            ]);
+        }
+
+        if (!$document->isEmpty()) {
+            if ($document->getAttribute('projectId') === $project->getId()) {
+                $resourceType = $document->getAttribute('resourceType');
+                $resourceId = $document->getAttribute('resourceId');
+                $message = "Domain already assigned to another resource.";
+                if (!empty($resourceId)) {
+                    $message .= " with ID '{$resourceId}'";
+                }
+
+                $message .= '.';
+            } else {
+                $message = 'Domain already assigned to different project.';
+            }
+
+            throw new Exception(Exception::RULE_ALREADY_EXISTS, $message);
+        }
+
+        $resourceInternalId = '';
+
+        switch ($resourceType) {
+            case 'function':
+            case 'site':
+                if (empty($resourceId)) {
+                    throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, '$resourceId cannot be empty for resourceType "' . $resourceType . '".');
+                }
+
+                $expectedDomain = ($resourceType === 'function') ? $functionsDomain : $sitesDomain;
+                if (!\str_ends_with($domain, $expectedDomain)) {
+                    throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'Domain must end with ' . $expectedDomain . ' for resourceType "' . $resourceType . '".');
```

Comment:
Doesn this deny any custom domains?

This endpoint should still accept `og.matejbaco.eu` domain

Reply:
Yes, but `resourceType` shouldn't be `site` or `function`. It should be `api`.